### ISSUE TITLE
Arg.parse prints error message multiple times

### ISSUE THREAD

**Issue by vicuna** [ocaml/ocaml#7578]
**Original bug ID:** 7578
**Reporter:** Richard-Degenne
**Assigned to:** @Octachron
**Status:** resolved (set by @Octachron on 2017-07-03T14:23:48Z)
**Resolution:** duplicate
**Priority:** normal
**Severity:** minor
**Version:** 4.04.2
**Fixed in version:** 4.05.0 +dev/beta1/beta2/beta3/rc1
**Category:** standard library
**Duplicate of:** #7523

## Bug description

When using the `Arg` module, a missing argument will trigger an error.

However, the error message is printed 3 times in a row instead of only once.

*Note* : I tried the same piece of code with 4.03.0 and 4.02.3, and was unable to reproduce.

## Steps to reproduce

With OCaml 4.04.2,

- Create a file `cli.ml` containing the following:

```
(* cli.ml *)

let usage = "usage: " ^ Sys.argv.(0) ^ " [-d int]"

let speclist = [
  ("-d", Arg.Set_int someint, ": some int parameter");
]

let () =
  Arg.parse
    speclist
    (fun x -> raise (Arg.Bad ("Bad argument : " ^ x)))
    usage;
  Printf.printf "%d\n" !someint
```

- Run with `$ ocaml cli.ml -d` (note the missing argument after -d)

- Look at the error message:

```
cli.ml: option '-d' needs an argument.
usage: cli.ml [-d int]
  -d : some int parameter
  -help  Display this list of options
  --help  Display this list of options
cli.ml: cli.ml: option '-d' needs an argument.
usage: cli.ml [-d int]
  -d : some int parameter
  -help  Display this list of options
  --help  Display this list of options
.
usage: cli.ml [-d int]
  -d : some int parameter
  -help  Display this list of options
  --help  Display this list of options
```g

**Comment by vicuna** [comment]
**Comment author:** Richard-Degenne

Crap, as soon as I file this, I find it's a duplicate of #7523.

Sorry. :(

**Comment by vicuna** [comment]
**Comment author:** @Octachron

This is fixed in 4.05, see https://github.com/ocaml/ocaml/blob/trunk/Changes#L455 and https://github.com/ocaml/ocaml/pull/999 for more information.

**Comment by vicuna** [comment]
**Comment author:** @Octachron

Thanks for reporting and finding the right duplicate id.
